lib/Extension/Debug/DebugExtension.php

```diff
@@ -27,25 +48,24 @@ public function load(ContainerBuilder $container): void
                 $container->getParameter(PhpactorContainer::PARAM_EXTENSION_CLASSES),
                 $container->get(DefinitionDocumentor::class)
             );
-        });
-
-        $container->register(RpcCommandDocumentor::class, function (Container $container) {
-            return new RpcCommandDocumentor(
-                $container->get('rpc.handler_registry'),
-                $container->get(DefinitionDocumentor::class)
-            );
-        });
+        }, [
+            self::TAG_DOCUMENTOR => ['name' => self::EXTENSION_DOCUMENTOR_NAME]
+        ]);
 
         $container->register('generate_documentation_command.extensions', function (Container $container) {
-            return new GenerateDocumentationCommand($container->get(ExtensionDocumentor::class));
+            return new GenerateDocumentationCommand(
+                $container->get(DocumentorRegistry::class)->get(self::EXTENSION_DOCUMENTOR_NAME)
+            );
         }, [
             ConsoleExtension::TAG_COMMAND => [
                 'name' => 'development:configuration-reference'
             ]
         ]);
 
         $container->register('generate_documentation_command.command', function (Container $container) {
-            return new GenerateDocumentationCommand($container->get(RpcCommandDocumentor::class));
+            return new GenerateDocumentationCommand(
+                $container->get(DocumentorRegistry::class)->get(RpcExtension::RPC_DOCUMENTOR_NAME)
```

Comment:
we need to inject the registry so that the documentors are _lazily_ instantiated (this is the main reason for having a registry). As mentioned before Symfony commands are pro-actively instantiated by default, and they are all instantiated when doing something like `phpactor list`, which means that all the documentors will be instantiated too.

Reply:
The phpactor list works. That's not the problem. The problem seems to be the test since the Debug extension now doesn't work in isolation anymore.

Reply:
I think we're not on the same page:

We inject the registry:

```
class FooCommand {
    public function __cosntruct(DocumentorRegistry $registry) {
        parent::__construct();
        $this->registry = $registry;
    }

    public function execite(InputArgument $input, // ...): int
    {
        $documentor = $this->registry->getByName($input->getArgument('documentor'));
        $documentor->document(); // etc
    }
}
```

the command does not know or care about what implementations it has, it just pulls a named documentor from the registry.